### ISSUE TITLE
Incorrect display on FF 66a / macOS

### ISSUE THREAD

**Issue by pabuisson** [marionebl/svg-term-cli#39]
Hey, 

Following up on [this](https://dev.to/pabuisson/comment/829h). Here's what I get when I display a term SVG in Firefox 66 (nightly), on macOS : 

![capture d ecran 2019-01-09 a 21 30 32](https://user-images.githubusercontent.com/769918/50926783-214dca80-1456-11e9-8581-a5464e4689ab.png)

The image is animated alright, but font size and spacing and everything seems broken. Same svg is displayed nicely on the same machine with a blink-based browser. I tried to turn off every ad-blocking and tracker-blocking and content-blocking addon and it's still the same. I also have an old FF 56 here on the same machine, and it works okay, I get the expected result.

**Comment by marionebl** [comment]
Thanks for reporting @pabuisson!

**Comment by marionebl** [comment]
@pabuisson:

Using FF 66.0a1 (2019-01-11) I could not reproduce the issue. Are all the svg images on Darren's post affected for you?

**Comment by pabuisson** [comment]
Yes all images. It might be an issue with an addon of mine, I'll try to find what could lead to this. Maybe in a few hours though. I'll keep you posted.

**Comment by pabuisson** [comment]
Found it! It was not related to any addon. Actually, in my Firefox, I got the "Minimum font size" set to 9. That's what's making the SVG looks like the screenshot of the first post. If I set "Minimum font size" to "None", then I get the expected result : 

![capture d ecran 2019-01-12 a 10 17 43](https://user-images.githubusercontent.com/769918/51071565-53784b80-1653-11e9-879d-bf41b7ee80df.png)

I checked and in my chrome-based browsers, I also have this "Minimum font size" set but it has no impact on the SVG rendering.

Hope it helped!

**Comment by marionebl** [comment]
Thank you for the analysis! I'll see what `svg-term-cli` can do to mitigate the issue.

**Comment by jwhitley** [comment]
This appears to be the current and correct tracking issue in Bugzilla.  Sadly there was a whole run of attempts to fix this, which all got backed out due to causing other intermittent regressions:

https://bugzilla.mozilla.org/show_bug.cgi?id=935056

But it does contain some info on what causes the regression, essentially Firefox applying minimum size clamping too early in SVG text size computation.  E.g. FF applies minimum size clamping to "5px" *in the SVG file's coordinate system*, not the actual on-screen rendered text size due factoring in the SVG viewport.  That suggests a possible solution where manipulating/scaling the coordinate system in the generated SVGs, but leaving the final on-screen size the same, may produce a fix.

**Comment by jwhitley** [comment]
For posterity's sake, I also found this comment on a nearly identical Chromium issue, before a fix was landed there:

https://bugs.chromium.org/p/chromium/issues/detail?id=232332#c11

> The minimal font size restriction can be bypassed in SVG by setting text-rendering="geometricPrecision" on the text element or parent elements[...]

Unfortunately, that attribute appears to have zero effect on the Firefox bug.  I manually edited the `svg-term-cli` parrot example to test this, but it's not a workaround.

**Comment by jwhitley** [comment]
See also [this SO post on this exact issue](https://stackoverflow.com/questions/55926730/how-to-override-minimum-font-size-in-firefox-affecting-text-within-svg/56329177).

**Comment by jonathanlermitage** [comment]
Thx for the `font.minimum-size` workaround! I faced this issue with Firefox 98, I can show animated SVG correctly now :partying_face:
